src/image_gen/imagegenutils.hpp

```diff
@@ -41,19 +42,25 @@
 namespace ovms {
 struct HttpPayload;
 std::variant<absl::Status, std::optional<resolution_t>> getDimensions(const std::string& dimensions);
-std::variant<absl::Status, std::optional<resolution_t>> getDimensions(const HttpPayload& payload);
+std::variant<absl::Status, std::optional<resolution_t>> getDimensions(const rapidjson::Document& payload);
+std::variant<absl::Status, std::optional<resolution_t>> getDimensions(const ovms::MultiPartParser& parser);
 
-std::variant<absl::Status, std::string> getPromptField(const HttpPayload& payload);
+std::variant<absl::Status, std::string> getPromptField(const rapidjson::Document& payload);
+std::variant<absl::Status, std::string> getPromptField(const ovms::MultiPartParser& payload);
 
-std::variant<absl::Status, std::optional<std::string>> getStringFromPayload(const ovms::HttpPayload& payload, const std::string& keyName);
-std::variant<absl::Status, std::optional<int64_t>> getInt64FromPayload(const ovms::HttpPayload& payload, const std::string& keyName);
-std::variant<absl::Status, std::optional<int>> getIntFromPayload(const ovms::HttpPayload& payload, const std::string& keyName);
-std::variant<absl::Status, std::optional<size_t>> getSizetFromPayload(const ovms::HttpPayload& payload, const std::string& keyName);
-std::variant<absl::Status, std::optional<float>> getFloatFromPayload(const ovms::HttpPayload& payload, const std::string& keyName);
+std::variant<absl::Status, std::optional<std::string>> getStringFromPayload(const rapidjson::Document& payload, const std::string& keyName);
```

Comment:
maybe this could be reused also in other future calulalators like stringutils. It could be considered a payloadutils